lib/meow/report.ex

```diff
@@ -210,10 +215,105 @@ defmodule Meow.Report do
         %{population: "population #{idx}", generation: population.generation}
       end
 
-    Vl.new(width: 300)
+    Vl.new(title: "Generations", width: 300)
     |> Vl.data_from_values(data)
     |> Vl.mark(:bar)
     |> Vl.encode_field(:x, "generation", type: :quantitative)
     |> Vl.encode_field(:y, "population", type: :nominal, axis: [title: nil])
   end
+
+  @doc """
+  Saves an HTML report including the summary and plots to a specified path.
+  """
+  @spec to_html(t(), Path.t()) :: :ok | {:error, File.posix()}
+  def to_html(report, path) do
+    assert_vega_lite!("to_html/2")
+
+    summary = format_summary(report)
+
+    plots = metrics_plots(report) ++ [plot_times(report)] ++ [plot_generations(report)]
+
+    plot_divs =
+      1..length(plots)
+      |> Enum.map(fn i -> ~s{<div id="plot-#{i}" class="plot"></div>} end)
+      |> Enum.join("\n")
+
+    script = plots_script(plots)
+
+    html = """
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <title>Meow Report</title>
+      <script src="https://cdn.jsdelivr.net/npm/vega@5.20.2"></script>
+      <script src="https://cdn.jsdelivr.net/npm/vega-lite@5.1.0"></script>
+      <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.17.0"></script>
+
+      <style>
+        .container {
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+          max-width: 900px;
+          margin: 0 auto;
+          padding: 1rem;
+        }
+
+        .header {
+          width: 100%;
+        }
+
+        .plot:not(:first-of-type) {
+          margin-top: 32px;
+        }
+
+        .summary {
+          white-space: pre-line;
+          width: 100%;
+        }
+      </style>
+    </head>
+    <body>
+      <div class="container">
+        <h1 class="header">Meow Report</h1>
+        <pre class="summary">#{summary}</pre>
+        #{plot_divs}
+      </div>
+      #{script}
+    </body>
+    </html>
+    """
+
+    File.write(path, html)
+  end
+
+  defp plots_script(plots) do
+    plots_json = Enum.map(plots, &VegaLite.Export.to_json/1)
+
+    vars =
+      plots_json
+      |> Enum.with_index()
+      |> Enum.map(fn {json, idx} ->
+        ~s{var spec#{idx + 1} = JSON.parse("#{escape_double_quotes(json)}");}
+      end)
+      |> Enum.join("\n")
+
+    embeds =
+      1..length(plots_json)
+      |> Enum.map(fn i -> ~s{vegaEmbed("#plot-#{i}", spec#{i});} end)
```

Comment:
You can totally do `vegaEmbed("...", JSON.parse(...))`, this should simplify things :D